hostap: Fix wrong security printing about WPA3 PWE

MaochenWang1 opened the pull request:
'wifi status' CMD shows wrong security information when STA connects to Ext-AP with WIFI_SECURITY_TYPE_SAE_HNP, after connection using WIFI_SECURITY_TYPE_SAE_AUTO. Setting sae_pwe for all the WPA3 SAE types can fix this issue.

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -1021,8 +1021,10 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		if (params->security == WIFI_SECURITY_TYPE_SAE_HNP ||
-		    params->security == WIFI_SECURITY_TYPE_SAE_H2E ||
-		    params->security == WIFI_SECURITY_TYPE_SAE_AUTO) {
+			params->security == WIFI_SECURITY_TYPE_SAE_H2E ||
+			params->security == WIFI_SECURITY_TYPE_SAE_AUTO) {
```
Please do not introduce no-op changes, the change looks actually wrong here becase indentation is now wrong.

MaochenWang1 replied:
Okay

krish2718 commented on modules/hostap/src/supp_api.c:
```diff
@@ -1045,14 +1047,23 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
-							   : 2)) {
-					goto out;
-				}
+			switch (params->security) {
+			case WIFI_SECURITY_TYPE_SAE_HNP:
+				sae_pwe = 0;
+				break;
+			case WIFI_SECURITY_TYPE_SAE_H2E:
+				sae_pwe = 1;
+				break;
+			case WIFI_SECURITY_TYPE_SAE_AUTO:
+				sae_pwe = 2;
+				break;
+			default:
+				sae_pwe = 0;
+				break;
+			}
+
+			if (!wpa_cli_cmd_v("set sae_pwe %d", sae_pwe)) {
+				goto out;
```
```
diff --git a/modules/hostap/src/supp_api.c b/modules/hostap/src/supp_api.c
index 1bd8edb857c..bf37e0af3fc 100644
--- a/modules/hostap/src/supp_api.c
+++ b/modules/hostap/src/supp_api.c
@@ -1045,12 +1045,15 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
                                }
                        }
 
-                       if (params->security == WIFI_SECURITY_TYPE_SAE_H2E ||
-                           params->security == WIFI_SECURITY_TYPE_SAE_AUTO) {
+                       if (params->security == WIFI_SECURITY_TYPE_SAE_HNP ||
+                               params->security == WIFI_SECURITY_TYPE_SAE_H2E ||
+                               params->security == WIFI_SECURITY_TYPE_SAE_AUTO) {
                                if (!wpa_cli_cmd_v("set sae_pwe %d",
                                                   (params->security == WIFI_SECURITY_TYPE_SAE_H2E)
                                                           ? 1
-                                                          : 2)) {
+                                                          : (params->security == WIFI_SECURITY_TYPE_SAE_AUTO)
+                                                                  ? 2
+                                                                  : 0)) {
                                        goto out;
                                }
                        }
```
Isn't this a simpler change?

MaochenWang1 replied:
Yes, thanks

MaochenWang1 committed:
hostap: Fix wrong security printing about WPA3 PWE

'wifi status' CMD shows wrong security information when STA connects to
Ext-AP with WIFI_SECURITY_TYPE_SAE_HNP, after connection using
WIFI_SECURITY_TYPE_SAE_AUTO. Setting sae_pwe for all the WPA3 SAE types
can fix this issue.

Signed-off-by: Maochen Wang <maochen.wang@nxp.com>
